### PR TITLE
SVCPLAN-3224: manage custom scripts

### PR DESCRIPTION
**Pull request by jakerundall**
Add custom_files param.

pdk update

pdk validate

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -38,11 +42,12 @@
   Hash              $crons,
   Array[String]     $dependencies,
   Integer           $logs_archive_after_days,
-  Optional[String]  $logs_archive_dir,
   Integer           $logs_compress_after_days,
   String            $logs_mgmt_script_path,
   String            $logs_path,
   Array[String]     $required_pkgs,
+  Hash              $custom_files = {},
```
Comment (billglick) [inline manifests/init.pp:49]: These 2 lines look odd to what I normally expect.

I've generally only assigned a default value here if a parameter is marked `Optional`. 

- Here the Optional parameter is not getting assigned a default value
- And the non-optional parameter is getting assigned a default

Perhaps that works, but is backwards to how I generally do it.

**Review by billglick — COMMENTED** [review]
The use of `Optional` parameters and default values for non-optional parameters seems off to me.

File: manifests/init.pp [hunk 2]
```diff
@@ -38,11 +42,12 @@
...
+  Optional[String]  $logs_archive_dir,
```
Comment (billglick) [inline manifests/init.pp]: These 2 lines look odd to what I normally expect.

I've generally only assigned a default value here if a parameter is marked Optional.

Here the Optional parameter is not getting assigned a default value
And the non-optional parameter is getting assigned a default

Perhaps that works, but is backwards to how I generally do it.

**Review by billglick — COMMENTED** [review]
The use of Optional parameters and default values for non-optional parameters seems off to me.

**Comment by jakerundall** [comment]
Thanks, Bill! I still don't fully get how these are supposed to work, but I think my update will make it align with what you've suggested, and consistency is good.

Now I've got just the one Optional logs_archive_dir with value = undef (and I pulled the null out of Hiera). custom_files is sort of ~optional still but I did that by putting {} in Hiera (because pdk validate complained if it was "Optional" and = {} right in the code).

Also reordered metadata param comments to be strictly alphabetical (rather than putting the optional params at the end, although the actual param defs list still has the Optional param at the end).

Is this better?

I'll squash before merging, but just keeping a separate commit for ease of review.

**Review by billglick — APPROVED** [review]
Yes, this looks better. Or at least aligns with my thinking on this.

Have you tested this with not supplying the `profile_acctd::logs_archive_dir`?

**Comment by jakerundall** [comment]
I hadn't checked as part of this update (we specify it for all acctd installation at this point AFAIK), but just did right now and it works as expected.

**Pull request merged**